lib/SILOptimizer/Mandatory/MandatoryInlining.cpp

```diff
@@ -413,14 +413,6 @@ static SILFunction *getCalleeFunction(
   if (!shouldInlinePredicate(AI, *CalleeFunction))
     return nullptr;
 
-  // If CalleeFunction is a declaration, see if we can load it.
-  if (CalleeFunction->empty())
```

Comment:
I moved these checks into the `shouldInlinePredicate` function, so that we can have different checks for TF and for non-TF inlining.

This is probably not the most correct solution, but it makes all the tests pass.

Reply:
If TF mandatory inlining needs to inline non-transaprent functions, please update the SIL linker in lib/SIL/Linker.cpp and lib/SILOptimizer/UtilityPasses/Link.cpp.

Reply:
Done: https://bugs.swift.org/browse/SR-8015

I'll also add a todo in the code here referencing the issue.